frontend/packages/console-app/src/components/cloud-shell/CloudShellExec.tsx

```diff
@@ -0,0 +1,159 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { Base64 } from 'js-base64';
+import { LoadingBox, StatusBox } from '@console/internal/components/utils';
+import { connectToFlags, WithFlagsProps } from '@console/internal/reducers/features';
+import { impersonateStateToProps } from '@console/internal/reducers/ui';
+import { FLAGS } from '@console/shared';
+import { WSFactory } from '@console/internal/module/ws-factory';
+import { resourceURL } from '@console/internal/module/k8s';
+import { PodModel } from '@console/internal/models';
+import { Terminal } from './AutoResizeTerminal';
+
+// pod exec WS protocol is FD prefixed, base64 encoded data (sometimes json stringified)
+
+// Channel 0 is STDIN, 1 is STDOUT, 2 is STDERR (if TTY is not requested), and 3 is a special error channel - 4 is C&C
+// The server only reads from STDIN, writes to the other three.
+// see also: https://github.com/kubernetes/kubernetes/pull/13885
+
+type CloudShellExecProps = {
+  container: string;
+  podname: string;
+  namespace: string;
+  shcommand?: string[];
```

Comment:
All of these props affect the web socket but none of them will cause the web socket to reconnect if they change.

We may not get into this use case for the current implementation however we should be creating reactive components that update according to their incoming props.